modules/products/goland/src/main/java/net/ashald/envfile/products/goland/GolandRunConfigurationExtension.java

```diff
@@ -24,8 +24,8 @@ protected String getEditorTitle() {
 
     @Override
     protected void patchCommandLine(@NotNull GoRunConfigurationBase<?> goRunConfigurationBase, @Nullable RunnerSettings runnerSettings, @NotNull GeneralCommandLine generalCommandLine, @NotNull String s) throws ExecutionException {
-        Map<String, String> currentEnv = generalCommandLine.getEnvironment();
-        Map<String, String> newEnv = EnvFileConfigurationEditor.collectEnv(goRunConfigurationBase, new HashMap<>(currentEnv));
+        Map<String, String> newEnv = EnvFileConfigurationEditor.collectEnv(goRunConfigurationBase, EnvUtil.getInitialEnv(generalCommandLine));
+        Map<String, String> currentEnv = generalCommandLine.getEffectiveEnvironment();
```

Comment:
`currentEnv ` seems unused?..

Reply:
Calling `getEffectiveEnvironment()` was wrong (unintentional change, missed in the code cleanup).
`cmdline.getEnvironment()` returns the reference to the environment, not a copy. As with the old code, `clear` and `putAll` on it modifies the data used by the command line. I added comments to clarify, because it's confusing at first sight.